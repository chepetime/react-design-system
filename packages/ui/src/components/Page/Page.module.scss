@use "./../../styles" as styles;
@use "~@chepe/tokens/tokens/tokens.scss" as tokens;

:root {
  --page-background-color: #{tokens.$base-gray-100};
  --page-text-color: #{tokens.$base-gray-900};
}

:global .dark {
  --page-background-color: #{tokens.$base-gray-900};
  --page-text-color: #{tokens.$base-gray-100};
}

.Page {
  @include styles.styles-reset;

  min-height: 100vh;

  overflow: auto;

  background-color: var(--page-background-color);
  color: var(--page-text-color);
}
