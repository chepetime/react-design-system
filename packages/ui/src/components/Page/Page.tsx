import * as React from "react";
import classnames from "classnames";
import css from "./Page.module.scss";

export interface PageProps {
  children?: React.ReactNode;
}

export const Page = React.forwardRef<HTMLDivElement, PageProps>(
  (props: PageProps, ref: React.Ref<HTMLDivElement>) => {
    return (
      <div className={classnames(css.Page)} ref={ref}>
        {props.children}
      </div>
    );
  }
);

Page.displayName = "Page";
