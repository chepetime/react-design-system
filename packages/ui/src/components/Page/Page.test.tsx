import React from "react";import { render } from "@testing-library/react";
import { Page } from "./Page";

describe("Page", () => {
  test("renders the Page component", () => {
    render(<Page />);
  });
});
