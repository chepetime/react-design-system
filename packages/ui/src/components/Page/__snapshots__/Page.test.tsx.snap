// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page matches snapshot 1`] = `
<div>
  <div
    class="Page"
  />
</div>
`;
